server/src/Application/Services/FlashcardImporter/FlashcardImporter.cs

```diff
@@ -22,21 +22,39 @@ public async Task<IReadOnlyCollection<CreateFlashcardDto>> ImportFlashcardsFromC
 
         var lines = stream.ReadLines();
         var importedFlashcards = new BlockingCollection<CreateFlashcardDto>();
+        var threadList = new List<Thread>();
+        var lockObject = new object(); 
 
-        Parallel.ForEach(lines, line =>
+        foreach (var line in lines)
         {
-            var values = line.Split(',');
-            if (values.Length != 2)
-            {
-                throw new ValidationException($"CSV file must only contain flashcard front and back values. Invalid values: {string.Join(',', values)}");
-            }
-
-            var front = values[0];
-            var back = values[1];
-            var createFlashcardDto = new CreateFlashcardDto(flashcardSetId, front, back);
+            var localLine = line; 
 
-            importedFlashcards.Add(createFlashcardDto);
-        });
+            var thread = new Thread(() =>
+            {
+                var values = localLine.Split(',');
+                if (values.Length != 2)
+                {
+                    throw new ValidationException($"CSV file must only contain flashcard front and back values. Invalid values: {string.Join(',', values)}");
+                }
+
+                var front = values[0];
+                var back = values[1];
+                var createFlashcardDto = new CreateFlashcardDto(flashcardSetId, front, back);
+
+                lock (lockObject)
+                {
+                    importedFlashcards.Add(createFlashcardDto);
```

Comment:
You don't need to lock the object since `importedFlashcards` is a thread-safe collection. Look into this article: https://learn.microsoft.com/en-us/dotnet/standard/collections/thread-safe/blockingcollection-overview